### PR TITLE
handle loading contracts error

### PR DESCRIPTION
**Pull request by 0oM4R**
### Description

- Added getGrid into try block
- Added an alert to show the error message
### Changes

if any error happen while listing the user contracts; An alert with the error message will appears, if the user click on refresh it will try to reconnect for like 1.5 min and will show the error.
this long time of reconnecting is probably related to the gridclient, will open an issue to improve this flow from the gridclient side 

[Screencast from 09 نوف, 2023 EET 12:51:16 م.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/79a76982-3a98-4853-8035-2635a252653e)

### Related Issues

#548 
### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by AlaaElattar** [comment]
- Please check conflicts.

**Comment by AlaaElattar** [comment]
- In this case what do u think if we showed only the error `Error listing contracts by twin id 86: AxiosError: Network Error` 
image

**Comment by 0oM4R** [comment]
> * In this case what do u think if we showed only the error `Error listing contracts by twin id 86: AxiosError: Network Error`
>   image

the only way to handle this case is to check on the error message, which is not recommended. later we will create error types on clients and we can handle each case, currently if we think we don't need the prefix will totally remove it from all cases

**Review by mohamedamer453 — APPROVED** [review]

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 1]
```diff
@@ -178,16 +182,22 @@ const headers: VDataTableHeader = [
...
   { title: "Details", key: "actions", sortable: false },
 ];
-
+const loadingError = ref("");
 async function onMount() {
+  loadingError.value = "";
   selectedContracts.value = [];
   loading.value = true;
   failedContractId.value = undefined;
   contracts.value = [];
```
Comment (zaelgohary) [inline packages/playground/src/weblets/tf_contracts_list.vue:191]: Can we reset these values in finally instead?

**Review by zaelgohary — CHANGES_REQUESTED** [review]

File: packages/playground/src/weblets/tf_contracts_list.vue (hunk 1 above)
Comment (0oM4R) [reply packages/playground/src/weblets/tf_contracts_list.vue:191]: we need to reset some values before starting, like hide the error message and remove field contracts ids, 
as we are not repeating this step inside catch, I think its fine to have them inside try block, what do you think ?

Comment (0oM4R) [reply packages/playground/src/weblets/tf_contracts_list.vue]: some of them must be in try like error message; if we reset it in finally , we will never have an error message

**Review by AlaaElattar — APPROVED** [review]

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**